### ISSUE TITLE
MOCKERY_CONFIG doesn't work the same as --config

### ISSUE THREAD

**Issue by alecxvs** [vektra/mockery#407]
I have a .mockery.yml file at my project root though my actual go module is in a subdirectory so I've been trying to set up some way to point mockery to the config file without copying the --config to every command usage (like putting go:generate in each test that uses a generated mock). I tried to set MOCKERY_CONFIG in the module's Makefile, but when running mockery, it seems to ignore the file's settings. `mockery showconfig` shows the config key is set correctly but the rest of the config is not. If I use --config directly though with the same value, it does set all the correct config.

In short, my observation:
MOCKERY_CONFIG sets `config` but does not load config from the file
--config sets `config` and loads config from the file

**Comment by LandonTClipp** [comment]
I'm not sure I've ever explicitly added a test for this. I wouldn't be surprised if this isn't working. 

One thing I've wanted to implement is behavior where mockery will look for `.mockery.yml` at every level of the cwd up to root. This way when you're using go generate, it will eventually find `.mockery.yml` at your project root.

PRs are welcome if you have the time.

**Comment by LandonTClipp** [comment]
@alecxvs I've fixed your issue in the linked PR.
